master/buildbot/db/compression/zstd.py

```diff
@@ -29,10 +30,36 @@ class ZStdCompressor(CompressorInterface):
     name = "zstd"
     available = HAS_ZSTD
 
+    COMPRESS_LEVEL = 9
+
     @staticmethod
     def dumps(data: bytes) -> bytes:
-        return zstandard.compress(data, level=9)
+        return zstandard.compress(data, level=ZStdCompressor.COMPRESS_LEVEL)
 
     @staticmethod
     def read(data: bytes) -> bytes:
-        return zstandard.decompress(data)
+        # not too sure why, but zstandard.decompress do not work with
+        # data compressed with `ZstdCompressionObj`
+        # `ZstdDecompressionObj` does and can also decompress data from `zstandard.compress`
+        decompress_obj = zstandard.ZstdDecompressor().decompressobj()
+        return decompress_obj.decompress(data) + decompress_obj.flush()
+
+    class CompressObj(CompressObjInterface):
```

Comment:
zstandard recomments against using CompressObjInterface due to performance reasons https://python-zstandard.readthedocs.io/en/latest/compressor.html#zstdcompressionobj . Was the streaming API proposed by them considered? Not suggesting to change anything in this PR because it's still an improvement.

Reply:
I interpreted the note as this API is for compatibility wit Python's stdlib compression APIs, so use it as such.
If possible, prefer other APIs.

Here, we could use `ZstdCompressionChunker`, and either implement the compress/flush calls joining the iterable returned.
Or change the Interface return type to `Generator[bytes, None, None]`, but that would change all implementation that do not benefit from it.

So I used this compatibility layer since it felt it made more sense for this use case.